astropy/units/core.py

```diff
@@ -1987,19 +1997,35 @@ def __init__(self, scale, bases, powers, decompose=False,
         # kwarg `_error_check` is False, the error checking is turned
         # off.
         if _error_check:
-            scale = sanitize_scale(scale)
             for base in bases:
                 if not isinstance(base, UnitBase):
                     raise TypeError(
                         "bases must be sequence of UnitBase instances")
             powers = [validate_power(p) for p in powers]
 
-        self._scale = scale
-        self._bases = bases
-        self._powers = powers
-        self._decomposed_cache = None
-        self._expand_and_gather(decompose=decompose, bases=decompose_bases)
-        self._hash = None
+        if not decompose and len(bases) == 1:
+            # Short-cut; with one unit there's no expand and gather.
+            unit = bases[0]
+            power = powers[0]
+            if power == 1:
+                scale *= unit.scale
+                self._bases = unit.bases
+                self._powers = unit.powers
+            elif power == 0:
+                self._bases = []
+                self._powers = []
+            else:
+                scale *= unit.scale ** power
+                self._bases = unit.bases
+                self._powers = [operator.mul(*resolve_fractions(p, power))
+                                for p in unit.powers]
+            self._scale = sanitize_scale(scale)
+        else:
+            self._scale = scale
```

Comment:
It looks like now the scale is only validated in the `if` clause above - is that ok?

Reply:
All inputs get adjusted by `_expand_and_gather`. Since that is far from obvious, I added a comment.